### PR TITLE
Response guidance

### PR DESCRIPTION
**Pull request by hamishtaplin**
## Changes
- Guidance now has show/hide toggle
- Added test for guidance
## Test
- `npm run test` 
- Confirm test passes
- Navigate to questionnaire and click on 'Show further guidance' toggle.
- Click again to close

### CODE REVIEW

File: app/js/app/modules/charlimit.js [hunk 1]
```diff
@@ -37,3 +44,5 @@ export default function() {
 
   return nodeList
 }
+
+domready(charLimit)
```
Comment (LJBabbage) [inline app/js/app/modules/charlimit.js]: Do you need domready and default function?

File: app/js/app/modules/domready.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+const EVENT_DOM_READY = 'DOMContentLoaded'
```
Comment (LJBabbage) [inline app/js/app/modules/domready.js]: Const in capital when the rest have been camel-case?

File: tests/karma/spec/domready.spec.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+describe('domready', () => {
+  let count = 1
+  domready(() => { count++ }) // 2
+  domready(() => { count++ }) // 3
+  domready(() => { count++ }) // 4
+
+  it('should execute a callback given once the dom is ready', () => {
+    expect(count).to.equal(4)
+  })
+})
```
Comment (LJBabbage) [inline tests/karma/spec/domready.spec.js:12]: Why are we counting to 4 on the dom being ready 3 times?

**Comment by LJBabbage** [comment]
I get a test failure 
 charlimit
    ✖ should return a nodeList with a length of 1

And toggle doesn't work when I run the application

File: tests/karma/spec/domready.spec.js (hunk 3 above)
Comment (hamishtaplin) [inline tests/karma/spec/domready.spec.js:12]: To test that it works concurrently, ie: more than once. Several modules will add callbacks that need to be called in succession.

File: app/js/app/modules/domready.js (hunk 2 above)
Comment (hamishtaplin) [inline app/js/app/modules/domready.js]: This is a convention for event name constants, probably not consistent with everything else though.

**Comment by hamishtaplin** [comment]
Weird, it works fine for me and on Travis. 

Try just running karma on it's own: `npm run karma`

File: app/js/app/modules/charlimit.js (hunk 1 above)
Comment (hamishtaplin) [inline app/js/app/modules/charlimit.js]: No, removed.

**Comment by dhilton** [comment]
I rebased this locally against master and got an issue with the test failing:

```
04 04 2016 12:29:37.329:INFO [framework.browserify]: bundle built
04 04 2016 12:29:37.343:INFO [karma]: Karma v0.13.18 server started at http://localhost:9876/
04 04 2016 12:29:37.346:INFO [launcher]: Starting browser PhantomJS
04 04 2016 12:29:38.332:INFO [PhantomJS 1.9.8 (Mac OS X 0.0.0)]: Connected on socket W5b6ZfWgeGoOOnHEAAAA with id 51513851
  charlimit
    ✔ should return a nodeList with a length of 1
    ✔ should render limit in to msg element
    ✔ should decrease the limit message by the number of characters entered
    ✔ should prevent excess characters being entered
  Details Toggle
    ✔ DOM should contain the template
    ✖ Should toggle class '.is-expanded' when clicked
  domready
    ✔ should execute a callback given once the dom is ready
Missing error handler on `socket`.
Error: Karma: Tests Failed
    at [object Object].<anonymous> (tests.js:26:13)
    at emitTwo (events.js:105:20)
    at [object Object].emit (events.js:185:7)
    at emitRunCompleteIfAllBrowsersDone (/Users/danhilton/projects/ONS/eq-survey-runner/node_modules/karma/lib/server.js:256:12)
    at [object Object].<anonymous> (/Users/danhilton/projects/ONS/eq-survey-runner/node_modules/karma/lib/server.js:278:9)
    at emitTwo (events.js:105:20)
    at [object Object].emit (events.js:185:7)
    at [object Object].onComplete (/Users/danhilton/projects/ONS/eq-survey-runner/node_modules/karma/lib/browser.js:142:13)
    at Socket.<anonymous> (/Users/danhilton/projects/ONS/eq-survey-runner/node_modules/karma/lib/events.js:13:22)
    at emitTwo (events.js:105:20)
    at Socket.emit (events.js:185:7)
    at Socket.onevent (/Users/danhilton/projects/ONS/eq-survey-runner/node_modules/socket.io/lib/socket.js:330:8)
    at Socket.onpacket (/Users/danhilton/projects/ONS/eq-survey-runner/node_modules/socket.io/lib/socket.js:290:12)
    at Client.ondecoded (/Users/danhilton/projects/ONS/eq-survey-runner/node_modules/socket.io/lib/client.js:193:14)
    at Decoder.Emitter.emit (/Users/danhilton/projects/ONS/eq-survey-runner/node_modules/component-emitter/index.js:134:20)
    at Decoder.add (/Users/danhilton/projects/ONS/eq-survey-runner/node_modules/socket.io-parser/index.js:247:12)
04 04 2016 12:29:48.438:WARN [PhantomJS 1.9.8 (Mac OS X 0.0.0)]: Disconnected (1 times), because no message in 10000 ms.
04 04 2016 12:29:48.440:ERROR [karma]: [Error: Karma: Tests Failed]
Error: Karma: Tests Failed
    at [object Object].<anonymous> (tests.js:26:13)
    at emitTwo (events.js:105:20)
    at [object Object].emit (events.js:185:7)
    at emitRunCompleteIfAllBrowsersDone (/Users/danhilton/projects/ONS/eq-survey-runner/node_modules/karma/lib/server.js:256:12)
    at [object Object].<anonymous> (/Users/danhilton/projects/ONS/eq-survey-runner/node_modules/karma/lib/server.js:278:9)
    at emitOne (events.js:95:20)
    at [object Object].emit (events.js:182:7)
    at [object Object]._onTimeout (/Users/danhilton/projects/ONS/eq-survey-runner/node_modules/karma/lib/browser.js:50:15)
    at Timer.listOnTimeout (timers.js:92:15)
[12:29:48] Finished 'test:scripts' after 24 s
[12:29:48] Starting 'test'...
[12:29:48] Finished 'test
```

I had to also install a "karma-chrome-launcher" manually - any ideas why this fails?

**Pull request merged**